modules/base/src/processors/transform.cpp

```diff
@@ -0,0 +1,163 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/base/processors/transform.h>
+
+namespace inviwo {
+
+namespace transform {
+
+TransformProperty::TransformProperty(const std::string& identifier, const std::string& displayName,
+                                     InvalidationLevel invalidationLevel,
+                                     PropertySemantics semantics)
+    : CompositeProperty(identifier, displayName, invalidationLevel, semantics) {}
+
+const std::string TranslateProperty::classIdentifier = "org.inviwo.trafo.TranslateProperty";
+std::string TranslateProperty::getClassIdentifier() const { return classIdentifier; }
+
+TranslateProperty::TranslateProperty(const std::string& identifier, const std::string& displayName,
+                                     const vec3& value, const vec3& minValue, const vec3& maxValue,
+                                     const vec3& increment, InvalidationLevel invalidationLevel,
+                                     PropertySemantics semantics)
+    : TransformProperty(identifier, displayName, invalidationLevel, semantics)
+    , translate("translate_", "Translate", value, minValue, maxValue, increment) {
+    translate.setSemantics(PropertySemantics::SpinBox);
+    addProperty(translate);
+}
+
+TranslateProperty::TranslateProperty(const TranslateProperty& rhs)
+    : TransformProperty(rhs), translate(rhs.translate) {
+    addProperty(translate);
+}
+
+TranslateProperty* TranslateProperty::clone() const { return new TranslateProperty(*this); }
+
+mat4 TranslateProperty::getMatrix() const { return glm::translate(*translate); }
+
+const std::string RotateProperty::classIdentifier = "org.inviwo.trafo.RotateProperty";
+std::string RotateProperty::getClassIdentifier() const { return classIdentifier; }
+
+RotateProperty::RotateProperty(const std::string& identifier, const std::string& displayName,
+                               const vec3& axisValue, const float angleValue, const float minAngle,
+                               const float maxAngle, const float increment,
+                               AngleMeasure angleMeasure, InvalidationLevel invalidationLevel,
+                               PropertySemantics semantics)
+    : TransformProperty(identifier, displayName, invalidationLevel, semantics)
+    , mode("mode", "Angle Measure",
+           {{"rad", "Radians", AngleMeasure::Radians}, {"deg", "Degrees", AngleMeasure::Degrees}})
+    , axis("axis", "Axis", glm::normalize(axisValue), vec3{-1.0f}, vec3{1.0f})
+    , angle("angle", "Angle", angleValue, minAngle, maxAngle, increment) {
+    mode.setSelectedValue(angleMeasure);
+    axis.setSemantics(PropertySemantics::SpinBox);
+    addProperties(mode, axis, angle);
```

Comment:
We have an angle property semantic as well if you want to use it.

Reply:
I think in this case a spinbox is better suited since it is more precise.